saltproc/depcode.py

```diff
@@ -159,20 +166,346 @@ def write_mat_file(self, dep_dict, dep_end_time):
+               Size of neutron population per cycle for Monte Carlo.
+           active_cycles : int, optional
+               Number of active cycles.
+           inactive_cycles : int, optional
+               Number of inactive cycles.
+
+        """
+        super().__init__("openmc",
+                         exec_path,
+                         template_inputfiles_path,
+                         geo_files,
+                         npop,
+                         active_cycles,
+                         inactive_cycles)
+        self.iter_inputfile = {'geometry': './geometry.xml',
+                               'settings': './settings.xml'},
+        self.iter_matfile = './materials.xml'
+
+    def read_depcode_info(self):
+        """Parses initial OpenMC simulation info from the OpenMC output files
+        and stores it in the `Depcode` object's ``sim_info`` attribute.
+        """
+
+    def read_depcode_step_param(self):
+        """Parses data from OpenMC depletion output for each step and stores
+        it in `Depcode` object's ``param`` attributes.
+        """
+
+    def read_dep_comp(self, read_at_end=False):
+        """Reads the depleted material data from the OpenMC depletion
+        simulation and returns a dictionary with a `Materialflow` object for
+        each burnable material.
+
+        Parameters
+        ----------
+        read_at_end : bool, optional
+            Controls at which moment in the depletion step to read the data.
+            If `True`, the function reads data at the end of the
+            depletion step. Otherwise, the function reads data at the
+            beginning of the depletion step.
+
+        Returns
+        -------
+        mats : dict of str to Materialflow
+            Dictionary that contains `Materialflow` objects.
+
+            ``key``
+                Name of burnable material.
+            ``value``
+                `Materialflow` object holding composition and properties.
+
+        """
+
+    def run_depcode(self, cores, nodes):
+        """Runs OpenMC depletion simulation as a subprocess with the given
+        parameters.
+
+        Parameters
+        ----------
+        cores : int
+            Number of cores to use for depletion code run.
+        nodes : int
+            Number of nodes to use for depletion code run.
+        """
+        # need to add flow control for plots option
+        args = (
+            'mpiexec',
+            '-n',
+            str(nodes),
+            'python',
+            './deplete_openmc.py'
+            '-mat',
+            self.iter_matfile,
+            '-geo',
+            self.iter_inputfile['geometry'],
+            '-set',
+            self.iter_inputfile['settings'],
+            '-tal',
+            self.iter_inputfile['tallies'],
+            '-dep',
+            self.iter_inputfile['depletion_settings'])
+
+        print('Running %s' % (self.codename))
+        # Need to figure out how to adapt this to openmc
```

Comment:
Is this comment #TODO? or did you leave it by accident?

Reply:
It is! I'll add that tag